src/ChangeFeedIterator.ts

```diff
@@ -0,0 +1,116 @@
+/// <reference lib="esnext.asynciterable" />
+import { isNumber } from "util";
+import { ChangeFeedOptions } from "./ChangeFeedOptions";
+import { Resource } from "./client";
+import { ClientContext } from "./ClientContext";
+import { Constants, ResourceType, StatusCodes } from "./common";
+import { FeedOptions } from "./request";
+import { Response } from "./request";
+
+export class ChangeFeedIterator<T> {
+  private static readonly IfNoneMatchAllHeaderValue = "*";
+  private nextIfNoneMatch: string;
+  private ifModifiedSince: string;
+  private lastStatusCode: number;
+  private isPartitionSpecified: boolean;
+
+  constructor(
+    private clientContext: ClientContext,
+    private resourceId: string,
+    private resourceLink: string,
+    private isPartitionedContainer: () => Promise<boolean>,
+    private changeFeedOptions: ChangeFeedOptions
+  ) {
+    // partition key XOR partition key range id
+    const partitionKeyValid = changeFeedOptions.partitionKey !== undefined;
+    const partitionKeyRangeIdValid = !(
```

Comment:
What's going on with this check? Does it mean passing partitionKeyRangeId === false is valid?

Reply:
That would be invalid, since it would be an invalid type. I should probably add a type check that it is a string. That would actually cover the other two cases, as well.